apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.cs

```diff
@@ -250,26 +289,18 @@ public static void ReleaseToPool(this SpannerClient client, Session session)
                 SessionPoolImpl targetPool = s_poolByClientAndDatabase.GetOrAdd(poolKey,
                                              key => new SessionPoolImpl(key));
 
-
-                SpannerClient evictionClient = poolKey.Client;
-                Session evictionSession = null;
-
                 //Figure out if we want to pool this released session.
-                lock (s_priorityListSync)
+                targetPool.ReleaseSessionToPool(client, session);
+                s_priorityList.Add(targetPool);
+                if (CurrentPooledSessions > MaximumPooledSessions)
```

Comment:
ReleaseToPool only guarantees that its net effect on poolsize will be zero if the cap is reached.  It doesnt guarantee the pool size will be within range when it returns.
The net effect is that the pool size should remain within range as long as its not manually changed.  If the developer resizes the pool to a lower value in production, the pool will shrink gradually as sessions get used and not repooled or otherwise evicted due to the eviction timer.